### ISSUE TITLE
Celestia's Cake Golf Adventure In Space: No Preloader

### ISSUE THREAD

**Issue by waspennator** [ruffle-rs/ruffle#9266]
### Describe the bug

[Celestia's Cake Golf Adventure In Space.zip](https://github.com/ruffle-rs/ruffle/files/10475447/Celestia.s.Cake.Golf.Adventure.In.Space.zip)
Preloader for the game doesn't show up at all and the swf spits this out in the console log.

Error dispatching event EventObject(EventObject { type: "progress", class: flash.events::ProgressEvent, ptr: 0x26681b0 }) to handler FunctionObject(FunctionObject { ptr: 0xfd9d50 }) : RustError("Cannot access property ::addEventListener of null or undefined")

### Expected behavior

There should be a loading bar popping up for a few seconds before transitioning to the title screen.

### Affected platform

Browser's extension

### Operating system

Windows 11

### Browser

Edge 109

**Comment by waspennator** [comment]
Retested again on the latest nightly and not only does the preloader appear now, but it does seem capable of getting ingame now. Granted it still has some visual issues. Other then that it seems to work now.

Flash:
![Celestia Flash](https://user-images.githubusercontent.com/78768499/220924716-0b8fbe56-37c2-4f33-bd03-f055e318087d.png)


On latest Ruffle Nightly:
![Celestia Ruffle2](https://user-images.githubusercontent.com/78768499/220924764-61720929-91a9-4b17-9de7-5b5670a717c8.png)

**Comment by waspennator** [comment]
![Celestia ruffle 3](https://github.com/ruffle-rs/ruffle/assets/78768499/5d43fb3d-0d84-49e7-b459-80be6efd4692)

Visuals have changed a bit, but now the game has regressed and crashes before getting to the golf part, trying it out on ruffle desktop, it either crarshes with

 panicked at 'either event handler is re-entrant (likely), or no event handler is registered (very unlikely)', C:\Users\runneradmin\.cargo\registry\src\index.crates.io-6f17d22bba15001f\winit-0.28.6\src\platform_impl\windows\event_loop\runner.rs:245:18

or

panicked at 'Error in Surface::get_current_texture_view: Validation Error

Caused by:
    Parent device is lost
', C:\Users\runneradmin\.cargo\registry\src\index.crates.io-6f17d22bba15001f\wgpu-0.16.1\src\backend\direct.rs:778:30

**Comment by waspennator** [comment]
Tried it on steam deck on web through edge, no crashes yet, but it does seem like it takes longer to load transitions like the title to the intro cutscene

**Comment by waspennator** [comment]
Can confirm on my s22 that it used to start instantly on older builds, but now freezes then crashes after a few seconds with this.

# Error Info
Error name: Error
Error message: panicked at 'wgpu error: Validation Error

Caused by:
    In Device::create_buffer
    Not enough memory left

', /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wgpu-0.16.1/src/backend/direct.rs:3019:5
Error stack:
```
Error: panicked at 'wgpu error: Validation Error

Caused by:
    In Device::create_buffer
    Not enough memory left

', /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wgpu-0.16.1/src/backend/direct.rs:3019:5
    at W.n.wbg.__wbg_new_d258248ed531ff54 (https://unpkg.com/@ruffle-rs/ruffle/core.ruffle.6abe6836b798bb12eac7.js:1:51548)
    at core::ops::function::FnOnce::call_once{{vtable.shim}}::h29556e60c8892742 (wasm://wasm/02cbe33e:wasm-function[8717]:0x851173)
    at std::panicking::rust_panic_with_hook::h85b7f6628c291e12 (wasm://wasm/02cbe33e:wasm-function[5671]:0x76d0a1)
    at core::panicking::panic_fmt::hf5c4cd929d4aaa9e (wasm://wasm/02cbe33e:wasm-function[8579]:0x84cceb)
    at core::ops::function::Fn::call::hf9acd6a8ee6ba6a0 (wasm://wasm/02cbe33e:wasm-function[5981]:0x784f3d)
    at wgpu::backend::direct::ErrorSinkRaw::handle_error::h464bdc1d764c25f7 (wasm://wasm/02cbe33e:wasm-function[4482]:0x700b83)
    at <T as wgpu::context::DynContext>::device_create_buffer::h4edf919a96e19857 (wasm://wasm/02cbe33e:wasm-function[6608]:0x7bad84)
    at <wgpu::Device as wgpu::util::device::DeviceExt>::create_buffer_init::hcb741d38bc6ad63c (wasm://wasm/02cbe33e:wasm-function[1441]:0x427b7a)
    at core::cell::once::OnceCell<T>::get_or_try_init::outlined_call::h621bef31221e9af4 (wasm://wasm/02cbe33e:wasm-function[5027]:0x735a1d)
    at ruffle_render_wgpu::surface::Surface::draw_commands_and_copy_to::hae28adf0a3b7102d (wasm://wasm/02cbe33e:wasm-function[1278]:0x3e3cfd)
```

# Player Info
Allows script access: false
Renderer: wgpu
Adapter Backend: Gl
Adapter Name: "ANGLE (Qualcomm, Adreno (TM) 730, OpenGL ES 3.2)"
Adapter Device Type: IntegratedGpu
Adapter Driver Name: ""
Adapter Driver Info: ""
Enabled features: Features(TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES | PUSH_CONSTANTS)
Available features: Features(TEXTURE_COMPRESSION_ETC2 | TEXTURE_COMPRESSION_ASTC | TEXTURE_COMPRESSION_ASTC_HDR | CLEAR_TEXTURE)
Current limits: Limits { max_texture_dimension_1d: 4096, max_texture_dimension_2d: 4096, max_texture_dimension_3d: 2048, max_texture_array_layers: 256, max_bind_groups: 4, max_bindings_per_bind_group: 640, max_dynamic_uniform_buffers_per_pipeline_layout: 8, max_dynamic_storage_buffers_per_pipeline_layout: 0, max_sampled_textures_per_shader_stage: 16, max_samplers_per_shader_stage: 16, max_storage_buffers_per_shader_stage: 0, max_storage_textures_per_shader_stage: 0, max_uniform_buffers_per_shader_stage: 11, max_uniform_buffer_binding_size: 16384, max_storage_buffer_binding_size: 0, max_vertex_buffers: 8, max_buffer_size: 268435456, max_vertex_attributes: 16, max_vertex_buffer_array_stride: 255, min_uniform_buffer_offset_alignment: 32, min_storage_buffer_offset_alignment: 256, max_inter_stage_shader_components: 60, max_compute_workgroup_storage_size: 0, max_compute_invocations_per_workgroup: 0, max_compute_workgroup_size_x: 0, max_compute_workgroup_size_y: 0, max_compute_workgroup_size_z: 0, max_compute_workgroups_per_dimension: 0, max_push_constant_size: 96 }
Surface quality: high
Surface samples: 4
Surface size: Extent3d { width: 550, height: 400, depth_or_array_layers: 1 }

# Page Info
Page URL: https://ruffle-regressions.github.io/?version=0.1.0-nightly.2023.6.28

# Browser Info
User Agent: Mozilla/5.0 (Linux; Android 10; K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Mobile Safari/537.36
Platform: Linux armv81
Has touch support: true

# Ruffle Info
Version: 0.1.0
Name: nightly 2023-06-28
Channel: nightly
Built: 2023-06-28T00:20:07.985Z
Commit: 9f67d3c23887170427a8e62c33436009168bbee8
Is extension: false

# Metadata
width: 960
height: 720
frameRate: 30
numFrames: 2
swfVersion: 10
backgroundColor: #000000
isActionScript3: true
uncompressedLength: 5903842

**Comment by waspennator** [comment]
Performance regression seems to be gone as of latest nightly and no longer seems to crash

**Comment by waspennator** [comment]
Should be fully resolved by #11878
